### PR TITLE
Make --yes work for task clear command

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Closes: https://github.com/apache/airflow/issues/14171

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -857,7 +857,8 @@ def clear(args):
                 task_regex=args.task_regex,
                 include_downstream=args.downstream,
                 include_upstream=args.upstream)
-
+    if args.yes:
+        args.no_confirm = args.yes
```
Comment (kaxil) [inline airflow/bin/cli.py:861]: Do we need a new argument for it?

We can just update L868 to:

```diff
- confirm_prompt=not args.no_confirm,
+ confirm_prompt=not args.yes
```

Comment (ephraimbuddy) [reply airflow/bin/cli.py:861]: I saw that no_confirm has been around for the command for a while, that made me not to remove it entirely. I will remove it now.

**Review by kaxil — COMMENTED** [review]
Actually yes - you are right. 

```

pytest command error: unrecognized arguments: -c, see help above.
___________________ CliTests.test_parentdag_downstream_clear ___________________

self = <tests.core.test_core.CliTests testMethod=test_parentdag_downstream_clear>

    def test_parentdag_downstream_clear(self):
        args = self.parser.parse_args([
>           'clear', 'example_subdag_operator.section-1', '--no_confirm'])

tests/core/test_core.py:1436: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.6/argparse.py:1737: in parse_args
    self.error(msg % ' '.join(argv))
airflow/bin/cli.py:4175: in error
    self.exit(2, '\n{} command error: {}, see help above.\n'.format(self.prog, message))
```

You can revert back the last commit, apologies for that

**Comment by ephraimbuddy** [comment]
> You can revert back the last commit, apologies for that
Ok

File: airflow/models/dag.py [hunk 2]
```diff
@@ -1189,7 +1189,7 @@ def clear_dags(
                           end_date=end_date,
                           only_failed=only_failed,
                           only_running=only_running,
-                          confirm_prompt=False,
+                          confirm_prompt=confirm_prompt,
```
Comment (kaxil) [inline airflow/models/dag.py]: This should be false, we already check confirm_prompt on L1178 and set `do_it` accordingly

```suggestion
                          confirm_prompt=False,
```

**Review by tooptoop4 — APPROVED** [review]

File: airflow/models/dag.py (hunk 2 above)
Comment (ephraimbuddy) [reply airflow/models/dag.py]: Done. Didn't read through the code well :)

**Review by kaxil — APPROVED** [review]

**Pull request merged**